__tests__/app.test.js

```diff
@@ -34,6 +35,65 @@ describe("/api/topics", () => {
 	});
 });
 
+describe("/api/articles", () => {
+	describe("GET", () => {
+		test("status:200, responds with an array of articles", () => {
+			return request(app)
+				.get("/api/articles")
+				.expect(200)
+				.then(response => {
+					const { articles } = response.body;
+					expect(articles).toBeInstanceOf(Array);
+					expect(articles).toHaveLength(12);
+					articles.forEach(article => {
+						expect(article).toEqual(
+							expect.objectContaining({
+								article_id: expect.any(Number),
+								title: expect.any(String),
+								topic: expect.any(String),
+								author: expect.any(String),
+								body: expect.any(String),
+								created_at: expect.any(String),
+								votes: expect.any(Number),
+								comment_count: expect.any(Number),
+							})
+						);
+					});
+				});
+		});
+
+		test("status:200, responds with an array of articles sorted by date in descending order", () => {
+			return request(app)
+				.get("/api/articles")
+				.expect(200)
+				.then(response => {
+					const { articles } = response.body;
+
+					expect(articles).toBeSortedBy("created_at", {
+						coerce: true,
+						descending: true,
+					});
+				});
+		});
+
+		test("status:200, responds with an array of articles which match the topic provided by a query of topic", () => {
+			return request(app)
+				.get("/api/articles?topic=mitch")
+				.expect(200)
+				.then(response => {
+					const { articles } = response.body;
+
+					expect(articles).toBeInstanceOf(Array);
+					expect(articles).toHaveLength(11);
+
+					articles.forEach(article => {
+						expect(article.topic).toBe("mitch");
+					});
+				});
+		});
+	});
+});
+
```

Comment:
Current tests are nice, but there are a few more tests that need to be added:
- Topic not found
- Topic exists, but has no articles

Think about how you'd differentiate between a topic not existing and a topic existing but also not having any articles 🧠